engine/src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -890,4 +880,14 @@ class SlickDataAccess(databaseConfig: Config) extends DataAccess {
 
     runTransaction(action)
   }
+
+  private def materializeWorkflowDescriptor(id: WorkflowId, workflowSourceFiles: WorkflowSourceFiles)(implicit ec: ExecutionContext): Future[WorkflowDescriptor] = {
```

Comment:
@Horneth Fixed w.r.t your comment, based on the existing design (i.e. using CromwellServer.MaterializeActor...`., haven't injected it's references down the chain yet until we agree on how to do that.